adr/adr-01-vfs/adr-01-vfs.md

```diff
@@ -0,0 +1,68 @@
+# ADR 1: Chain and VFS Path Referencing
+
+## Changelog
+
+- 7DEC2024: Intial draft
+
+## Status
+
+Proposed
+
+## Context
+
+ADOs will need to be able to reference resources locally and on other chains. It needs to be human readable, standardized, and flexible enough for future expansion as the protocol grows. The primary usage of this schema will be for aOS kernel consumption. Optionally, other protocols can use the schema including offchain environments such as a web browser or local command line.
+
+## Decision
+
+Adopt the already standardized URI ([Uniform Resource Indicator](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier_)). We have borrowed many standard from the POSIX frameworks that have been in use since the 1960s and are ubiquitious across all systems worldwide.
+
+![image](https://upload.wikimedia.org/wikipedia/commons/thumb/d/d6/URI_syntax_diagram.svg/2136px-URI_syntax_diagram.svg.png)
+
+- scheme = andr://
+- userinfo = aOS username
+- host = blockchain (Andromeda Chain being the authority, for now)
+- port = n/a for now (\*)
+- path = vfs handled by host VFS registry
+- query = execution or query parameters
+- fragment = n/a for now (\*)
+
+(\*) These are reserved for future use and declared for backwards compatibility.
+
+Examples:
+
+- andr://osmosis/home/alice/splitter?count
+- andr://andromeda/etc/blocktime
+- andr://cosmos/chain/validators/valoper1
+- andr://archway/ibc/channels/12?destination
+
+## Consequences
+
+### ADO/Kernel Updates
+
+A retooling of a ADOs for standardizing this schema may be needed. The parsing and execution of these URIs will be done by the kernel, not the ADO itself.
+
+### Resolvers
+
+Resolvers will need to be established to ensure look ups. For readonly, off chain needs, a local resolver can be used to query a trusted caching/indexing system (graphql) or can be read from a local chain client.
+
+As an example an HTML anchor href (\<a href=""\>) links to an andr:// resource. The andr:// handler could be a web resolver such as [https://andr.zone](https://andr.zone) where a quick tranlation happens:
+
+`andr://osmosis/home/alice/splitter?count` -> `https://osmosis.andr.zone/home/alice/splitter?count` would return either a JSON object or HTML landing page.
```

Comment:
Same as above, should this not be `andr://alice@osmosis/home/splitter?count`?

Reply:
We should probably get on a call. Think about how those URIs are used with Unix systems.

ftp://user:password@hostname/path/to/object

In this case, password wouldn't ever be used, but maybe some sorta access token could be generated. I'm thinking advanced permissioning in the future.